src/components/common/Main.js

```diff
@@ -0,0 +1,133 @@
+import React, { Component } from 'react';
+import clsx from 'clsx';
+import PropTypes from 'prop-types';
+import { withTranslation } from 'react-i18next';
+import { connect } from 'react-redux';
+import MenuIcon from '@material-ui/icons/Menu';
+import { withStyles } from '@material-ui/core/styles';
+import ChevronRightIcon from '@material-ui/icons/ChevronRight';
+import { CssBaseline, Fab } from '@material-ui/core';
+import { toggleSideMenu } from '../../actions';
+import Header from './Header';
+import SideMenu from './SideMenu';
+import { DEFAULT_HEADER_VISIBLE, DRAWER_WIDTH } from '../../config/constants';
+
+const styles = (theme) => ({
+  root: {
+    display: 'flex',
+    height: '100%',
+  },
+  fullScreen: {
+    width: '100%',
+    height: '100%',
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  content: {
+    flexGrow: 1,
+    transition: theme.transitions.create('margin', {
+      easing: theme.transitions.easing.sharp,
+      duration: theme.transitions.duration.leavingScreen,
+    }),
+  },
+  contentShift: {
+    transition: theme.transitions.create('margin', {
+      easing: theme.transitions.easing.easeOut,
+      duration: theme.transitions.duration.enteringScreen,
+    }),
+    marginRight: DRAWER_WIDTH,
+  },
+  drawerHeader: {
+    display: 'flex',
+    alignItems: 'center',
+    padding: theme.spacing(0, 1),
+    ...theme.mixins.toolbar,
+    justifyContent: 'flex-end',
+  },
+  fab: {
+    left: theme.spacing(2),
+    top: theme.spacing(2),
+    position: 'absolute',
+    zIndex: 999999,
```

Comment:
Can we factor this out to a constant? Like `MAXIMUM_Z_INDEX` or something like that.